### DISCUSSION TITLE
Getting openvino open source 2022/1.1 to work on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by rogerjames99** [openvinotoolkit/openvino#18127] [Q&A]
I am trying to get openvino 2022/1.1 to work on Ubuntu 22.0 and talk to my NCS2 stick I have run into a problem. This is where I have got to so far.

1. Cloned the openvino repo.
2. Checked out the releases/2022/2.2 branch.
3. Initialised the submodules.
4. The build encountered a number compiler warnings that were treated as errors.
5. I fixed the error in src/core/src/type/bfloat16.cpp.
`@@ -57,7 +57,7 @@ size_t bfloat16::size() const {
 }
 
 bfloat16::operator float() const {
-    uint32_t tmp = (static_cast<uint32_t>(m_value) << 16);
+    static uint32_t tmp = (static_cast<uint32_t>(m_value) << 16);
     const float* f = reinterpret_cast<const float*>(&tmp);
     return *f;
 }`

6. I realised all the other errors were in a template in single include file  and that it would not be easy to fix. So I turned off the treat warnings as errors flag in the CMake config.
7. I stuck my head in a bucket of cold water.
8. I got as far as building the python wheels and hit a problem with the numpy version. So I took a chance and modified src/bindings/python/requirements.txt to remove the top limit on the numpy version. This caused an error in src/bindings/python/wheel/CMakeLists.txt.

`@@ -33,7 +33,7 @@ endforeach()
 
 execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "import wheel.bdist_wheel ; print(f'{wheel.bdist_wheel.get_abi_tag()}')" OUTPUT_VARIABLE PYTHON_ABI)
 execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "import wheel.vendored.packaging.tags as tags ; print(f'{tags.interpreter_name()}{tags.interpreter_version()}')" OUTPUT_VARIABLE INTERPRETER)
-execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "import wheel.vendored.packaging.tags as tags ; print(f'{next(tags._platform_tags())}')" OUTPUT_VARIABLE WHEEL_PLATFORM)
+execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "import wheel.vendored.packaging.tags as tags ; print(f'{next(tags.platform_tags())}')" OUTPUT_VARIABLE WHEEL_PLATFORM)
 string(STRIP ${PYTHON_ABI} PYTHON_ABI)
 string(STRIP ${INTERPRETER} INTERPRETER)
 string(STRIP ${WHEEL_PLATFORM} WHEEL_PLATFORM)
`

9. This finally got everything to build.
10. But as I expected there was a runtime error!

`Python 3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from openvino.runtime import Core
/home/roger/.local/lib/python3.10/site-packages/openvino/runtime/utils/types.py:24: FutureWarning: In the future `np.bool` will be defined as the corresponding NumPy scalar.
  (Type.boolean, np.bool),
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/roger/.local/lib/python3.10/site-packages/openvino/runtime/__init__.py", line 38, in <module>
    from openvino.runtime.ie_api import Core
  File "/home/roger/.local/lib/python3.10/site-packages/openvino/runtime/ie_api.py", line 16, in <module>
    from openvino.runtime.utils.types import get_dtype
  File "/home/roger/.local/lib/python3.10/site-packages/openvino/runtime/utils/types.py", line 24, in <module>
    (Type.boolean, np.bool),
  File "/home/roger/.local/lib/python3.10/site-packages/numpy/__init__.py", line 313, in __getattr__
    raise AttributeError(__former_attrs__[attr])
AttributeError: module 'numpy' has no attribute 'bool'.
`np.bool` was a deprecated alias for the builtin `bool`. To avoid this error in existing code, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations. Did you mean: 'bool_'?
>>> 
`

Numpy strikes again. 

Has anyone been down this path before?
Has anyone got any ideas on how to get past with minimal intervention in the source?

Help!

Roger

**Comment by berrylizzard** [comment]
Maybe this will help. I've gotten OpenVINO 2022.2 working on Ubuntu 22.04. Following (somewhat) the guide from [here](https://docs.openvino.ai/latest/openvino_docs_install_guides_installing_openvino_from_archive_linux.html) to install runtime, then `pip install openvino-dev[tensorflow2]==2022.2.0` for the dev toolkit. Note, that 2022.2.0 supports Python 3.6-3.9 so I'm using Python 3.9.16. I also made sure to have numpy (v1.23.1) installed before these steps.
Here's how:
1. Download file _l_openvino_toolkit_ubuntu20_2022.2.0.7713.af16ea1d79a_x86_64.tgz_ (94.6 MB) from [archive](https://storage.openvinotoolkit.org/repositories/openvino/packages/2022.2/linux/) and rename it for ease of use to openvino_2022.2.0.tgz
2. Unzip it and move to a known directory (I created an `~/intel/` folder)
```
$ tar -xf openvino_2022.2.0.tgz
$ mv l_openvino_toolkit_ubuntu20_2022.2.0.7713.af16ea1d79a_x86_64 ~/intel/openvino_2022.2.0
$ cd ~/intel/openvino_2022.2.0/install_dependencies
```
3. Download the _install_openvino_dependencies.sh_ file from [here](https://github.com/openvinotoolkit/openvino/tree/2022.3.0/scripts/install_dependencies). 
Rename it to _install_openvino_dependencies_2022.3.0.sh_ and move it to `~/intel/openvino_2022.2.0./install_dependencies/` (to avoid “Unsupported OS: ubuntu22.04” error; [2022.3.0 does support Ubuntu 22.04](https://www.intel.com/content/www/us/en/developer/articles/release-notes/openvino-2022-3-lts-relnotes.html)).
4.
```
$ chmod +x install_openvino_dependencies_2022.3.0.sh
$ sudo -E ./install_openvino_dependencies_2022.3.0.sh
$ cd 
$ source intel/openvino_2022.2.0/setupvars.sh
```
5. Check:
```
$ python -c "from openvino import runtime; print(runtime.get_version())"
2022.2.0-7713-af16ea1d79a-releases/2022/2
```
6. Now install dev toolkit ([reference](https://docs.openvino.ai/2022.3/openvino_docs_install_guides_install_dev_tools.html)) with `pip install openvino-dev[tensorflow2]==2022.2.0` and check that it works with `mo -h`.

**Comment by rogerjames99** [comment]
Thanks berrylizard,

Your info will be really useful. I had literally just started down the same path by cloning a fresh copy of the repo and checking out the releases/2202/3 branch. I have just checked and the build of openvino has finished succesfully. I will now follow your instructions for the dev toolkit.

Roger

**Reply by rogerjames99** [reply]
I got a succesful build with the following.

git clone https://github.com/openvinotoolkit/openvino.git myvino2
cd myvino2
git checkout releases/2022/3
mkdir build
python -m venv build/vino
build/vino/bin/activate
git submodule update --init --recursive
cd build
ccmake -DCMAKE_BUILD_TYPE=Debug ..
make -j$(nproc --all)
pip install openvino-dev[tensorflow2]==2022.3

But it is not seeing my NCS2 stick.

(vino) roger@dragon:~/myvino2/build$ python
Python 3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import openvino.runtime as ov
>>> core = ov.Core()
>>> print (core.available_devices)
['CPU']
>>>

I set the log level to trace.

export OPENVINO_LOG_LEVEL=5

But no logging was visible. I have not got a clue how this logging system is supposed to work or where the output goes to. Any help on this would be gratefully received.

As far as I know the myriad udev rules are set up correctly. The device is visible in lsusb. It also works as expected when mapped into an Ubuntu 20.04 vm running the 2022/1.1 release.

Roger
